swap_meet/vendor.py

```diff
@@ -1,2 +1,167 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    """A class representing a vendor who wants to swap items with other vendors
+
+    Attributes:
+        inventory (list): a list of items to swap with other vendors.
+    """
+
+    def __init__(self, inventory = None):
+        """Initializes the vendor with given inventory or set to None as default.
+
+        Args:
+            inventory(list): a list of items            
+        """
+        self.inventory = [] if inventory is None else inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Swaps items between the current vendor and another vendor.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap items with.
+            my_item (Item): The item the current vendor wants to trade.
+            their_item (Item): The item the other vendor wants to trade.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+
+        """
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.remove(their_item)
+
+        other_vendor.add(my_item)
+        self.add(their_item)
+
+        return True
```

Comment:
Not at all required, but rather a neat little trick: The remove method you created will return the item, which we could then append into our list all at once if we wanted to!

```suggestion
        other_vendor.add(self.remove(my_item))
        self.add(other_vendor.remove(their_item))

        return True
```